### PR TITLE
 Fix #2295: Add countdown timer for the challenges

### LINKED ISSUE #2295: Add countdown timer for the challenges
**Current Behaviour**:

There is no timer shown when the challenge deadline is approaching.

**Expected Behaviour**:

There should be a counter which will automatically appear one month before the challenge deadline.

Reference: #1769

### PR DESCRIPTION
**Pull request by Sanji515**
- Used the library [flipclockjs](http://flipclockjs.com/) for the countdown timer
- Created a wrapper for FlipClock.js jQuery
- Written the custom CSS for the countdown timer

Here's the screencast of the working

Recording #14

### CODE REVIEW

**Comment by Sanji515** [comment]
@RishabhJain2018 @galipremsagar, please review

**Comment by galipremsagar** [comment]
Looks nice, will check it out. Can you update your branch? Looks like yours is out-of-date.

**Comment by Sanji515** [comment]
@galipremsagar done, thanks.

File: frontend/src/js/controllers/challengeCtrl.js [hunk 1]
```diff
@@ -1805,4 +1816,28 @@
...
+            return {
+                replace: true,
+                template: '<div></div>',
+                restrict: 'E',
+                link: function(scope, element) {
+                    setTimeout(function() {
+                        if (countDownTimer <= 0 || countDownTimer == 'moreThanOneMonth') {
+                            countDownTimer = 0;
+                            element.css("display", "none");
+                        }else {
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeCtrl.js]:
```suggestion
                        } else {
```

File: frontend/src/js/controllers/challengeCtrl.js [hunk 2]
```diff
@@ -1805,4 +1816,28 @@
...
+                            angular.element('#challenge-logo').css('height', '130px');
+                        }
+
+                        element.FlipClock(countDownTimer, {
```
Comment (galipremsagar) [inline frontend/src/js/controllers/challengeCtrl.js]: Screenshot 2019-05-09 at 5 12 45 AM

Please fix this.

Comment (galipremsagar) [reply frontend/src/js/controllers/challengeCtrl.js]: Due to this, I'm not able to see the clock.

Comment (Sanji515) [reply frontend/src/js/controllers/challengeCtrl.js]: @galipremsagar Can you please clean navigator cache

File: frontend/src/js/controllers/challengeCtrl.js (hunk 1 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeCtrl.js]: @galipremsagar done

File: frontend/src/js/controllers/challengeCtrl.js (hunk 2 above)
Comment (Sanji515) [reply frontend/src/js/controllers/challengeCtrl.js]: @galipremsagar fixed this error, please check

**Comment by Sanji515** [comment]
> There is an error, please check.

@galipremsagar resolved, please check.

@RishabhJain2018 @deshraj Please review

**Comment by RishabhJain2018** [comment]
I think we should have the timer on every phase rather than on a challenge as most challenges last for unlimited time for some phase while some phase ends soon.
Also, please refer this for the timer https://aideadlin.es/?sub=ML,CV,NLP,RO,SP

**Pull request closed without merging**